pkg/workflow/asset/data/schemas/v1/Workflow.json

```diff
@@ -0,0 +1,94 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Workflow",
+  "description": "A project nebula workflow definition",
+  "type": "object",
+  "properties": {
+    "apiVersion": {
+      "type": "string",
+      "description": "Nebula workflow schema version",
+      "enum": [
+        "v1"
+      ]
+    },
+    "description": {
+      "type": "string",
+      "description": "An optional user-provided workflow description"
+    },
+    "parameters": {
+      "type": "object",
+      "patternProperties": {
+        "(.*?)": {
+          "$ref": "#/definitions/Parameter"
+        }
+      },
+      "additionalProperties": false
+    },
+    "steps": {
+      "type": "array",
+      "description": "List of workflow steps",
+      "items": {
+        "$ref": "#/definitions/Step"
+      }
+    }
+  },
+  "required": [
+    "apiVersion",
+    "steps"
+  ],
+  "definitions": {
+    "Parameter": {
+      "type": "object",
+      "description": "The parameters required to run this workflow",
+      "properties": {
+        "default": {
+          "description": "Default parameter value"
+        },
+        "description": {
+          "type": "string",
+          "description": "Optional parameter description"
+        }
+      }
+    },
+    "Step": {
+      "type": "object",
+      "description": "A workflow step definition",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique workflow step name"
+        },
+        "image": {
+          "type": "string",
+          "description": "Docker image for step execution"
+        },
+        "dependsOn": {
+          "type": "array",
+          "description": "A list of step dependencies",
+          "items": {
+            "type": "string"
+          }
+        },
+        "command": {
+          "type": "string",
+          "description": "Command to issue"
+        },
+        "args": {
+          "type": "array",
+          "description": "Command arguments",
+          "items": {
+            "type": "string"
+          }
+        },
+        "input": {
+          "type": "array",
+          "description": "Input script to execute",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": ["name", "image"]
```

Comment:
I wonder whether we should implement the new spec changes for approvals now, or wait until next sprint/iteration when it's fully baked? If we decide to validate now, this will require a bit of refactoring to support the different types.